Update making-proposals.mdx (#5962)
dfinity/portal: docs/building-apps/governing-apps/managing/making-proposals.mdx

```

*5. Submit the `UpgradeSnsControlledCanister` proposal to upgrade the SNS-controlled canister using the chunks.*
You may use any frontend for creating SNS proposals, or do it via the CLI. For example:

```
dfx canister call $SNS_GOVERNANCE_CANISTER_ID manage_neuron '(
 record {
   subaccount = blob "proposer-subaccount";
   command = opt variant {
     MakeProposal = record {
       url = "https://example.com";
       title = "Test chunked upgrades";
       action = opt variant {
         UpgradeSnsControlledCanister = record {
           new_canister_wasm = blob "";
           mode = opt (1 : int32);
           canister_id = opt principal "'${TARGET}'";
           chunked_canister_wasm = opt record {
             wasm_module_hash = blob "ABC";
             chunk_hashes_list = vec {
               blob "abc";
               ...
               blob "abc";
             };
             store_canister_id = opt principal "'${STORE}'";
           };
           canister_upgrade_arg = opt blob "XYZ";
         }
       };
       summary = "Test chunked upgrades";
     }
   };
 },
)'

```

### `ManageDappCanisterSettings`

> This proposal allows upgrading the settings of one or more SNS DAO-controlled dapp canisters. The canisters whose settings should be updated are specified in `canister_ids`, and the proposal allows updating the freezing threshold, the reserved cycles limit, the log visibility (which can be visible to just controllers or the public), the memory allocation, the compute allocation, and the wasm memory threshold.

Reviewer: ```suggestion
This proposal allows upgrading the settings of one or more SNS DAO-controlled dapp canisters. The canisters whose settings should be updated are specified in `canister_ids`, and the proposal allows updating the freezing threshold, the reserved cycles limit, the log visibility (which can be visible to just controllers or the public), the memory allocation, the compute allocation, and the Wasm memory threshold.
```

#### Relevant type signatures

```candid
type ManageDappCanisterSettings = record {
  freezing_threshold : opt nat64;
  canister_ids : vec principal;
  reserved_cycles_limit : opt nat64;
  log_visibility : opt int32;
  wasm_memory_limit : opt nat64;
  memory_allocation : opt nat64;
  compute_allocation : opt nat64;
  wasm_memory_threshold : opt nat64;
};
enum LogVisibility {
  LOG_VISIBILITY_UNSPECIFIED = 0;
  LOG_VISIBILITY_CONTROLLERS = 1;
  LOG_VISIBILITY_PUBLIC = 2;
}
```

> Example in bash:

Reviewer: ```suggestion
Example using quill:
```

```bash
quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
    record {
        title = "Set the Memory Allocation of the Widget Canister to 314_159 Bytes";
        url = "https://sns-examples.com/proposal/42";
        summary = "Set the memory allocation of the Widget Canister to 314_159 bytes.";
        action = opt variant {
            ManageDappCanisterSettings = record {
                canister_ids = vec { principal "WIDGET-CANISTER-ID" };
                memory_allocation = opt 314_159;
            }
        }
    }
)' --canister-ids-file <PATH_TO_CANISTER_IDS_JSON_FILE> > message.json

quill send message.json
```

### `AdvanceSnsTargetVersion`

> As explained on in [Framework and Architecture](https://learn.internetcomputer.org/hc/en-us/articles/34140764336788-Framework-and-Architecture), all approved SNS canister versions are stored on the NNS canister SNS-W.

Reviewer: ```suggestion
As explained in the [Framework and Architecture](https://learn.internetcomputer.org/hc/en-us/articles/34140764336788-Framework-and-Architecture) article, all approved SNS canister versions are stored on the NNS canister SNS-W.
```

New SNS canister Wasm codes are approved by NNS proposals and then added to SNS-W.
An SNS community can then specify by proposal that they would like to follow the upgrade path up to a specific version.

:::info
To simplify maintenance further for SNS communities, an SNS community can also set `automatically_advance_target_version` in the nervous system parameters to true. In that case, the SNS will automatically upgrade the SNS to the newest available version as soon as it is approved by the NNS community.
:::

#### Relevant type signatures
```candid
type AdvanceSnsTargetVersion = record {
  new_target : opt SnsVersion;
};
```

### `UpgradeSnsToNextVersion`

:::warning
While this proposal can still be used to update the SNS framework canisters, it is recommended to use `AdvanceSnsTargetVersion` explained above instead, which is the newer version of this.
:::

> As explained on in [Framework and Architecture](https://learn.internetcomputer.org/hc/en-us/articles/34140764336788-Framework-and-Architecture), all approved SNS canister versions are stored on the NNS canister SNS-W.

Reviewer: ```suggestion
As explained in the [Framework and Architecture](https://learn.internetcomputer.org/hc/en-us/articles/34140764336788-Framework-and-Architecture) article, all approved SNS canister versions are stored on the NNS canister SNS-W.
```

New SNS canister Wasm codes are approved by NNS proposals and then added to SNS-W.
Each SNS community can then simply decide if and when they want to upgrade their SNS instance to the next SNS version that is available on SNS-W.

> To do so, they can use an `UpgradeSnsToNextVersion` proposal. If this proposal is adopted, it will trigger a call to SNS root that will ask SNS-W which new wasm version is available and then upgrade to this new version.

Reviewer: ```suggestion
To do so, they can use an `UpgradeSnsToNextVersion` proposal. If this proposal is adopted, it will trigger a call to the SNS root that will ask the SNS-W canister which new Wasm version is available, then trigger an upgrade to that version.
```

#### Relevant type signatures

```candid
    type UpgradeSnsToNextVersion : record {};
```

#### Putting it together

> Example in bash:

Reviewer: ```suggestion
Example using quill:
```

quill send message.json
```

## Custom proposals

Each SNS community might have functions that they would like to only execute if the SNS DAO agrees on it, but that might be very dapp-specific. Custom proposals, also called generic proposals, generic functions, or generic nervous system functions, allow a flexible way for SNS communities to define such functions.

For additional motivation and examples, also refer to the [Learn Hub](https://learn.internetcomputer.org/hc/en-us/articles/34146571133204-Proposals#h_01K0V8PJ2MV9T0YNCZ12ZZFRA9).

> Typically a generic proposal will have the following structure: a developer sends a proposal to add, execute, remove, or assign a topic to a custom proposal, i.e., a "generic nervous system function". So even though the proposal types below are technically "native proposal types," they are used to manage generic proposals.

Reviewer: ```suggestion
Typically a generic proposal will have the following structure: a developer sends a proposal to add, execute, remove, or assign a topic to a custom proposal, i.e., a "generic nervous system function". Therefore, even though the proposal types below are technically "native proposal types," they are used to manage generic proposals.
```

See more [security best practices](/docs/building-apps/security/overview).

### Handling generic proposals

**To use a generic proposal, it first needs to be added to the SNS governance system.** This means that the SNS DAO needs to approve that this is a proposal that should be supported going forward. As you have seen, generic proposals also have security implications; it is important to have this explicit approval.

Generic proposals can then also be removed again from SNS governance if they are not needed anymore.

To use a generic proposal, i.e., submit such a proposal, one uses the “execute generic nervous system function” proposal type and specifies which of the registered generic proposals should be used. Next it is explained how to submit each of these proposals.

> Finally, all SNS proposal types have assigned topics that are used for filtering and for vote delegation. Since generic proposals are SNS-specific, each SNS community must define which topic fits a new generic proposal best. This is done when the generic proposal is added. Later the community can re-assign a proposal to a new topic.

Author: I will optimistically close this comment as I think it is better to merge this and if we want to add something follow up with another PR as this is already more precise than what we have

> Finally, all SNS proposal types have assigned topics that are used for filtering and for vote delegation. Since generic proposals are SNS-specific, each SNS community must define which topic fits a new generic proposal best. This is done when the generic proposal is added. Later the community can re-assign a proposal to a new topic.

Reviewer 2: ```suggestion
Finally, all SNS proposal types have assigned topics that are used for filtering and for vote delegation. Since custom proposals are SNS-specific, each SNS community must define which topic fits best a particular generic proposal. This is done when the generic proposal is registered. Later, the community can re-assign a proposal to a new topic.
```

> ### Register a new generic proposal

Author: @aterga , for the case study, I think the example would not work anymore as now one also has to define a topic for each proposal. Would it be easy to add a line to the example to fix this? (otherwise we can just remove the concrete example for now, but would of course be nicer to have it

Author: Removed the example for now as I think it is better to merge and add this back later when we had the time to review
